src/sentry/models/apitoken.py

```diff
@@ -28,10 +29,98 @@ def default_expiration():
     return timezone.now() + DEFAULT_EXPIRATION
 
 
-def generate_token():
+def generate_token(token_type: AuthTokenType | str | None = AuthTokenType.__empty__) -> str:
+    if token_type:
+        return f"{token_type}{secrets.token_hex(nbytes=32)}"
+
     return secrets.token_hex(nbytes=32)
 
 
+class PlaintextSecretAlreadyRead(Exception):
+    def __init__(
+        self,
+        message="the secret you are trying to read is read-once and cannot be accessed directly again",
+    ):
+        super().__init__(message)
```

Comment:
Today I learned that this is even valid. :D 

```python
class PlaintextSecretAlreadyRead(Exception):
    """the secret you are trying to read is read-once and cannot be accessed directly again"""
    pass


class NotSupported(Exception):
    """the method you called is not supported by this token type"""
    pass
```

Reply:
Because python is whitespace based, you just need something to let python know the class definition has ended; which is why the doc string works just fine haha